extensions/agroal/deployment/src/main/java/io/quarkus/agroal/deployment/AgroalProcessor.java

```diff
@@ -551,89 +551,91 @@ void registerMetrics(
             boolean metricsEnabledForThisDatasource = dataSourcesBuildTimeConfig.metricsEnabled &&
                     aggregatedDataSourceBuildTimeConfig.getJdbcConfig().enableMetrics.orElse(true);
             Tag tag = new Tag("datasource", DataSourceUtil.isDefault(dataSourceName) ? "default" : dataSourceName);
+            // Metric classes expect null as the name for the default DS
+            String dataSourceNameOrNull = DataSourceUtil.isDefault(dataSourceName) ? null : dataSourceName;
```

Comment:
Sorry I missed that but... could we adjust that, pass the name to the gauge and counter and use `DataSourceUtil.isDefault()` there?

That would be more consistent and avoid this sort of issues.

Reply:
Sounds like a good idea. Done